File: frontend/src/api/pricingBatch.ts
import request from './request'

export interface PricingBatchCreateRequest {
  productIds: number[]
  strategyGoal: string
  constraints: string
}

export interface PricingBatchCreateResponse {
  batchId: number
  batchCode: string
  totalCount: number
  linkedTaskIds: number[]
  createFailedCount: number
}

export interface PricingBatchDetail {
  batchId: number
  batchCode: string
  batchStatus: string
  strategyGoal: string
  constraintText?: string
  totalCount: number
  runningCount: number
  completedCount: number
  manualReviewCount: number
  failedCount: number
  cancelledCount: number
  finalizedAt?: string | null
  createdAt: string
  updatedAt: string
}

export interface PricingBatchItem {
  id: number
  batchId: number
  itemOrder: number
  productId: number
  taskId?: number | null
  resultId?: number | null
  productTitle: string
  currentPrice?: number | null
  finalPrice?: number | null
  expectedSales?: number | null
  expectedProfit?: number | null
  profitGrowth?: number | null
  creationStatus: string
  taskStatus?: string | null
  displayStatus: string
  executeStrategy?: string | null
  reviewRequired?: boolean
  appliedStatus?: string | null
  errorMessage?: string | null
  createdAt: string
  batchItemUpdatedAt?: string | null
  taskUpdatedAt?: string | null
  updatedAt: string
}

export interface PricingBatchItemsPage {
  records: PricingBatchItem[]
  total: number
  current: number
  size: number
}

export interface PricingBatchListPage {
  records: PricingBatchDetail[]
  total: number
  current: number
  size: number
}

export interface PricingBatchCancelResponse {
  cancelledCount: number
  skippedCount: number
}

export const createPricingBatch = (data: PricingBatchCreateRequest) => {
  return request.post('/pricing/batches', data)
}

export const getRecentPricingBatches = (params: { page?: number; size?: number; status?: string } = {}) => {
  return request.get('/pricing/batches', { params })
}

export const getPricingBatchDetail = (batchId: number) => {
  return request.get(`/pricing/batches/${batchId}`)
}

export const getPricingBatchItems = (batchId: number, params: { page?: number; size?: number; status?: string }) => {
  return request.get(`/pricing/batches/${batchId}/items`, { params })
}

export const cancelPricingBatch = (batchId: number) => {
  return request.post(`/pricing/batches/${batchId}/cancel`)
}

File: backend-java/src/main/java/com/example/pricing/entity/PricingBatch.java
package com.example.pricing.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 批量定价主任务实体，对应一次批量定价批次。
 */
@Data
@TableName("pricing_batch")
public class PricingBatch {
    @TableId(type = IdType.AUTO)
    private Long id;

    @TableField("batch_code")
    private String batchCode;

    @TableField("requested_by_user_id")
    private Long requestedByUserId;

    @TableField("strategy_goal")
    private String strategyGoal;

    @TableField("constraint_text")
    private String constraintText;

    @TableField("total_count")
    private Integer totalCount;

    @TableField("completed_count")
    private Integer completedCount;

    @TableField("manual_review_count")
    private Integer manualReviewCount;

    @TableField("failed_count")
    private Integer failedCount;

    @TableField("cancelled_count")
    private Integer cancelledCount;

    @TableField("batch_status")
    private String batchStatus;

    @TableField("finalized_at")
    private LocalDateTime finalizedAt;

    @TableField(value = "created_at", fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(value = "updated_at", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
